services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -6386,6 +6387,64 @@ func TestUpdateDatadogMetricsInternal(t *testing.T) {
 	}
 }
 
+func TestDatadogQueueMetric(t *testing.T) {
+	tcs := []struct {
+		Name           string
+		changes        *TransformerResult
+		transformers   []Transformer
+		expectedGauges int
+	}{
+		{
+			Name: "Changes are sent as one event",
+			transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "envA",
+					Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Latest: true}},
+				},
+				&CreateApplicationVersion{
+					Application: "app1",
+					Manifests: map[string]string{
+						"envA": "envA-manifest-1",
+					},
+					WriteCommitData: false,
+				},
+				&CreateApplicationVersion{
+					Application: "app2",
+					Manifests: map[string]string{
+						"envA": "envA-manifest-2",
+					},
+					WriteCommitData: false,
+				},
+			},
+			expectedGauges: 2,
+		},
+	}
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			//t.Parallel() // do not run in parallel because of the global var `ddMetrics`!
+			ctx := WithTimeNow(testutil.MakeTestContext(), time.Unix(0, 0))
+			var mockClient = &MockClient{}
+			var client statsd.ClientInterface = mockClient
+			ddMetrics = client
+			repo := setupRepositoryTest(t)
+
+			err := repo.Apply(ctx, tc.transformers...)
+
+			if err != nil {
+				t.Fatalf("Expected no error: %v", err)
+			}
+
+			if tc.expectedGauges != len(mockClient.gauges) {
+				// Don't compare the value of the gauge, only the number of gauges (concurrency, values may change!, creating a flakey test)
```

Comment:
Why is this a problem?
The values should not change at this point in the test.
It seems to work fine for TestUpdateDatadogMetricsInternal.
I would prefer if we do the test the same way here.

Reply:
The problem is the value of the gauge. There is no way of knowing what the exact value of the queue size will be. It might be that when we gauge the size of the queue after inserting, the concurrent process extracting the elements from the queue already removed the element, thus making the value of the gauged queue size 0 when we insert a new element. It might also happend that an element has not yet been extracted, making the value 1.

The point is: The value of the first gauge cannot safely be assumed to be 1.

Reply:
The other tests do not use the queue (they use the ApplyTransformersInternal method) so they can safely assume the values of the gauges.

Reply:
ok, then let's just rephrase it a little, eg.: 
```suggestion
				// Don't compare the value of the gauge, only the number of gauges,
				// because we cannot be sure at this point what the size of the queue was during measurement
```